src/Frame.cc

```diff
@@ -91,7 +91,7 @@ Frame::Frame(uint32_t session_id, carta::FileLoader* loader, const std::string&
     _num_stokes = (_stokes_axis >= 0 ? _image_shape(_stokes_axis) : 1);
 
     // load full image cache for loaders that don't use the tile cache and mipmaps
-    if (!(_loader->UseTileCache() && _loader->HasMip(2)) && !FillImageCache()) {
+    if (!(_loader->UseTileCache() && _loader->HasMip(2)) && !FillImageCache(CurrentStokes())) {
```

Comment:
I suggest making Stokes an optional parameter which defaults to `CURRENT_STOKES` and is set to `CurrentStokes()` _inside_ the function. We could then avoid changing this everywhere.

It would also be cleaner to write accessor functions for the image cache, mutexes, etc., which have the same behaviour. Then instead of using `_image_caches[CurrentStokes()]` everywhere we could just write `ImageCache()` for the default or `ImageCache(some_other_stokes)`.